CCDFDataModel/CCDFVariable.h

```diff
@@ -291,21 +291,29 @@ namespace CDF {
     void setSize(size_t size) { currentSize = size; }
     size_t getSize() { return currentSize; }
 
-    Attribute *getAttribute(const char *name) const {
+    bool hasAttribute(const char *name) const {
       for (size_t j = 0; j < attributes.size(); j++) {
         if (attributes[j]->name.equals(name)) {
-          return attributes[j];
+          return true;
         }
       }
-      throw(CDF_E_ATTNOTFOUND);
-      return NULL;
+      return false;
+    }
+
+    Attribute *getAttribute(const char *name) const {
+      Attribute *a = getAttributeNE(name);
+      if (a == nullptr) {
+        throw(CDF_E_ATTNOTFOUND);
+      }
+      return a;
     }
     Attribute *getAttributeNE(const char *name) const {
-      try {
-        return getAttribute(name);
-      } catch (int e) {
-        return NULL;
+      for (size_t j = 0; j < attributes.size(); j++) {
+        if (attributes[j]->name.equals(name)) {
```

Comment:
CLion suggests this, but you might not want to make this change now:

```suggestion
      for (auto attribute : attributes) {
        if (attribute->name.equals(name)) {
```

Reply:
We should refactor this class altogether in another PR. Also the throw stuff is costly and could be done with a return statement.